pkg/server/ca/ca_test.go

```diff
@@ -77,6 +80,35 @@ func (s *CATestSuite) TestNoX509CASet() {
 	s.Require().EqualError(err, "X509 CA is not available for signing")
 }
 
+func (s *CATestSuite) TestSignX509SVID() {
+	subject := pkix.Name{
+		Country:      []string{"IGNORE ME"},
+		Organization: []string{"IGNORE ME"},
+	}
+
+	svidChain, err := s.ca.SignX509SVID(ctx, s.generateCSRWithSubject("example.org", subject), X509Params{})
+	s.Require().NoError(err)
+	s.Require().Len(svidChain, 1)
+
+	svid := svidChain[0]
+
+	s.False(svid.NotBefore.IsZero(), "NotBefore is not set")
+	s.False(svid.NotAfter.IsZero(), "NotAfter is not set")
+	s.NotEmpty(svid.SubjectKeyId, "SubjectKeyId is not set")
+	s.Equal(x509.KeyUsageKeyEncipherment|x509.KeyUsageKeyAgreement|x509.KeyUsageDigitalSignature, svid.KeyUsage, "key usage does not match")
+	s.Equal([]x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth}, svid.ExtKeyUsage, "ext key usage does not match")
+	s.False(svid.IsCA, "CA bit is set")
+	s.True(svid.BasicConstraintsValid, "Basic constraints are not valid")
+
+	// SPIFFE ID should be set to that of the trust domain
+	if s.Len(svid.URIs, 1, "has no URIs") {
+		s.Equal("spiffe://example.org", svid.URIs[0].String())
```

Comment:
Should we be allowing this? Perhaps we should file an issue, I think it is against SPIFFE spec